edx_solutions_api_integration/courses/views.py

```diff
@@ -2199,75 +2171,44 @@ class CoursesMetricsSocial(MobileListAPIView):
     ### The CoursesMetricsSocial view allows clients to query about the activity of all users in the
     forums
-    - URI: ```/api/users/{course_id}/metrics/social/?organization={org_id}```
-    - GET: Returns a list of social metrics for users in the specified course. Results can be filtered by organization
+    - URI: ```/api/courses/{course_id}/metrics/social/?organization={org_id}&score={bool}```
+    - GET: Returns a list of social metrics for users in the specified course.
+        * use `organization` query param to filter users by organization
+        * use `score` query param to get only scores of the users
     """
 
     def get(self, request, course_id):  # pylint: disable=arguments-differ
         if not request.user.is_staff:
             return Response({}, status=status.HTTP_401_UNAUTHORIZED)
 
-        total_enrollments = 0
-        data = {}
-
-        try:
-            slash_course_id = get_course_key(course_id, slashseparated=True)
-            # the forum service expects the legacy slash separated string format
-
-            organization = request.query_params.get('organization', None)
-            attribute_keys = css_param_to_list(self.request, 'attribute_keys')
-            attribute_values = css_param_to_list(self.request, 'attribute_values')
-
-            # load the course so that we can see when the course end date is
-            course_descriptor, course_key, course_content = get_course(self.request, self.request.user, course_id)  # pylint: disable=W0612,C0301
-            if not course_descriptor:
-                raise Http404
+        if not course_exists(course_id):
+            raise Http404
 
-            # get the course social stats, passing along a course end date to remove any activity after the course
-            # closure from the stats
-            data = get_course_social_stats(slash_course_id, end_date=course_descriptor.end)
+        organization = request.query_params.get('organization', None)
+        scores = request.query_params.get('scores', False)
+        course_key = get_course_key(course_id)
+        # remove any excluded users from the aggregate
+        exclude_users = get_aggregate_exclusion_user_ids(course_key)
 
-            course_key = get_course_key(course_id)
+        if scores:
+            data = StudentSocialEngagementScore.get_course_engagement_scores(course_key, organization, exclude_users)
 
-            # remove any excluded users from the aggregate
-            exclude_users = get_aggregate_exclusion_user_ids(course_key)
+        else:
+            data = StudentSocialEngagementScore.get_course_engagement_stats(course_key, organization, exclude_users)
 
-            for user_id in exclude_users:
-                if str(user_id) in data:
-                    del data[str(user_id)]
-            enrollment_qs = CourseEnrollment.objects.users_enrolled_in(course_key).filter(is_active=True)\
+            enrollment_qs = CourseEnrollment.objects.users_enrolled_in(course_key) \
```

Comment:
Shouldn't this be outside of the if/else?

Reply:
I didn't find any place in Apros where scores would be needed along with enrollments. I added the alternative behavior to this endpoint to return only scores.